### PR TITLE
feat(NativeCtorsGenerator): Generate nested types properly

### LINKED ISSUE #7014: NativeCtorsGenerator fails for nested classes
### Current behavior

In UWP, the following works:

```csharp
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            Content = new MyNestedControl();
        }

        public partial class MyNestedControl : Button
        {
            public MyNestedControl()
            {
                Foreground = new SolidColorBrush(Windows.UI.Color.FromArgb(255, 255, 0, 0));
                Content = "Click me!";
            }
        }
    }
```

For Android, it generates invalid code:

```csharp
partial class MainPage.MyNestedControl
```

### Expected behavior

Generated code should be correct:

```csharp
partial class MainPage
{
    partial class MyNestedControl
    {
    }
}
```

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #7014

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs [hunk 1]
```diff
@@ -6,18 +6,100 @@
...
+		/// 
+		/// while (disposables.Count > 0)
+		/// {
+		///     disposables.Pop().Dispose();
+		/// }
+		/// ]]></code>
+		/// <para>NOTE: Another possible implementation is to accept an <see cref="Action"/> as a parameter to generate the type members, execute the action here, and also dispose
+		/// the stack here. The advantage is that callers don't need to worry about disposing the stack.</para>
+		/// </remarks>
+		public static Stack<IDisposable> AddToIndentedStringBuilder(this INamedTypeSymbol namedTypeSymbol, IIndentedStringBuilder builder, Action<IIndentedStringBuilder>? beforeClassHeaderAction = null)
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs:42]: Hopefully, this should make it easy for other generators to prepare the boilerplate code.

**Review by Youssef1313 — COMMENTED** [review]
Note: `DependencyObjectGenerator` may need to be updated as well. It already has its own mechanism for nested classes, but pretty sure it can fail some test cases. For example, it doesn't seem to be handling generics, interfaces, structs.

These are not common, but is a good-to-have. Let me know if you want me to fix it in a follow up PR.

File: src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs [hunk 2]
```diff
@@ -6,18 +6,123 @@
...
+		/// {
+		///     builder.AppendLineInvariant("Console.WriteLine(\"Hello world\")");
+		/// }
+		/// 
+		/// while (disposables.Count > 0)
+		/// {
+		///     disposables.Pop().Dispose();
+		/// }
+		/// ]]></code>
+		/// <para>NOTE: Another possible implementation is to accept an <see cref="Action"/> as a parameter to generate the type members, execute the action here, and also dispose
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs:61]: Let me know if you want me to do this. It shouldn't be hard to do.

**Comment by jeromelaban** [comment]
> Note: `DependencyObjectGenerator` may need to be updated as well. It already has its own mechanism for nested classes, but pretty sure it can fail some test cases. For example, it doesn't seem to be handling generics, interfaces, structs.
> 
> These are not common, but is a good-to-have. Let me know if you want me to fix it in a follow up PR.

That should do for now. It'd be good to also add an actual generation of a nested dependency object in the runtime tests assembly, so that it's validated outside of the generator tests themselves.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**